moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -332,26 +341,23 @@ moveit::core::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const
           }
 
           std::string ksolver_timeout_param_name = base_param_name + ".kinematics_solver_timeout";
+          rclcpp::Parameter ksolver_timeout_param = declare_parameter(node_, ksolver_timeout_param_name);
           if (node_->has_parameter(ksolver_timeout_param_name))
           {
-            double ksolver_timeout;
-            rclcpp::Parameter parameter = node_->get_parameter(ksolver_timeout_param_name);
-            if (parameter.get_type() == rclcpp::ParameterType::PARAMETER_DOUBLE)
+            if (ksolver_timeout_param.get_type() == rclcpp::ParameterType::PARAMETER_DOUBLE)
```

Comment:
Shouldn't we just declare the parameter as double? That way we don't have to check the different cases

Reply:
It will throw the following exception `rclcpp::exceptions::InvalidParameterTypeException` if the loaded parameter is integer